### ISSUE TITLE
Publishers' sort order differs between `Customize` dialog and sidebar-filter listing

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#35167]
## Steps to Reproduce

1. install `1.63.93`
2. launch Brave
3. open a new-tab page
4. click on `Turn on Brave News`
5. add 30 or more publishers
6. compare the sort-order between the `Customize` dialog's listing and the sidebar-filter listing

## Actual result:

`Customize` dialog | sidebar-filter listing
--------------------|--------------------
<img width="322" alt="Screenshot 2024-01-06 at 1 10 31 AM" src="https://github.com/brave/brave-browser/assets/387249/a80a3d4b-dfe3-4aa0-9ee8-a45ba5f00402"> | <img width="259" alt="Screenshot 2024-01-06 at 1 10 50 AM" src="https://github.com/brave/brave-browser/assets/387249/b74008ca-2ae4-4ef5-8043-539bbb321107">

## Reproduces how often: 

100%

## Brave version (brave://version info)

```
Brave | 1.63.96 Chromium: 120.0.6099.199 (Official Build) nightly (x86_64)
-- | --
Revision | 566b736710f9d37f44176dad8aa6cc56df392294
OS | macOS Version 14.3 (Build 23D5043d)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @martin-c0des @fallaciousreasoning @mattmcalister @LorenzoMinto @brave/qa-team

**Comment by fallaciousreasoning** [comment]
Yeah, it's kinda weird - the sidebar on the feed is ordered by number of visits (in the last 3 weeks) while the order in the customize dialog is alphabetical.

These two sorts were deliberate at the time (@LorenzoMinto @martin-c0des) but it is odd having them different.

**Comment by LorenzoMinto** [comment]
It is a bit odd, I agree. Although they are "optimized" for the two separate use cases: in the feed you want to have handy the ones you visit the most, in the customise you want to quickly find "Source X" to unfollow it. Thoughts @martin-c0des @mattmcalister?

Also, @martin-c0des rightly noted that those lists are redundant now, the one in the customise might be removed in the upcoming settings redesign.
